app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -661,15 +673,10 @@ public void onResume() {
         presenter.attachView(this);
         registerNetworkReceiver();
         if (isResumed() && ((MainActivity) getActivity()).activeFragment == ActiveFragment.NEARBY) {
-            if (!isPermissionDenied && !applicationKvStore.getBoolean(
-                "doNotAskForLocationPermission", false)) {
-                if (!locationManager.isGPSProviderEnabled()) {
-                    startMapWithCondition("Without GPS");
-                } else {
```

Comment:
From this snippet, it seems we were invoking `startMapWithCondition` when the device didn't have GPS enabled. Right now I suppose we only invoke the similar flow when the location permission is not granted. Should we also not cover the case of invoking `startMapWithoutPermission` when GPS is turned off. If it is somehow take care already, could you clarify how?

Reply:
In the current flow inside `onViewCreated()`, `recenterMap()` is called by `presenter.setActionListeners(applicationKvStore)`, which in turn calls the locationPermissionGranted(), which checks if the GPS is on/off and performs actions accordingly. Thus, I didn't need to have a different condition for the GPS state of on/off.

Reply:
Ok. This might get tricky if at all `onResume` method would invoked more often in the fragment lifecycle than the `onViewCreated` method. Hopefully things don't get that tricky as that.